### PR TITLE
Itertools

### PR DESCRIPTION
**Pull request by Jongy**

### CODE REVIEW

**Comment by adrian17** [comment]
(For simplicity, couldn't we use the `itertools` implementation from ouroboros? It'd probably have a higher compatibility than we have now too.)

File: vm/src/stdlib/itertools.rs [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+    fn iter(zelf: PyRef<Self>, _vm: &VirtualMachine) -> PyRef<Self> {
+        zelf
+    }
+}
+
+pub fn make_module(vm: &VirtualMachine) -> PyObjectRef {
+    let ctx = &vm.ctx;
+
+    let count = ctx.new_class("count", ctx.object());
+    PyItertoolsCount::extend_class(ctx, &count);
```
Comment (coolreader18) [inline vm/src/stdlib/itertools.rs]: This can be `PyItertoolsCount::make_class(ctx)` if you put `#[pyclass(name = "count")]` on the struct declaration.

**Comment by windelbouwman** [comment]
@adrian17 this was also my first idea, but it seems a lot of modules are implemented in C in cpython, so we could do the same with rustpython.

File: vm/src/stdlib/itertools.rs (hunk 1 above)
Comment (Jongy) [reply vm/src/stdlib/itertools.rs]: I'll make use of this trick in the following PRs, thanks :)

**Pull request merged**